backend/api/serializers/ComplianceReport.py

```diff
@@ -209,8 +210,14 @@ class ComplianceReportWorkflowStateSerializer(serializers.ModelSerializer):
     )
 
     def should_show(self, field_name, value):
-        user = self.context['request'].user \
-            if 'request' in self.context else None
+        user = cache.get("user")
```

Comment:
This use of caching a user is a security concern because the user would be accessible globally and could be picked up by a different user using the system. Please revert these bits of caching code and get the user from the request context.